### PR TITLE
Search rooms

### PR DESCRIPTION
**Pull request by phucynwa**

### CODE REVIEW

**Comment by phucynwa** [comment]
ready

File: app/assets/javascripts/custom.js [hunk 1]
```diff
@@ -10,32 +10,13 @@ $(document).ready(function () {
     $('html, body').animate({scrollTop: 0}, 600);
     return false;
   });
+  $('#select-category').change(function() {
+      $('#search-rooms').submit();
```
Comment (mdmahmudurrahman) [inline app/assets/javascripts/custom.js]: Indentation

File: app/assets/javascripts/custom.js [hunk 2]
```diff
@@ -10,32 +10,13 @@ $(document).ready(function () {
...
+    }
+  );
 
+  $('#select-floor').change(function() {
+      $('#search-rooms').submit();
```
Comment (mdmahmudurrahman) [inline app/assets/javascripts/custom.js]: The same here

File: app/controllers/rooms_controller.rb [hunk 3]
```diff
@@ -2,7 +2,18 @@ class RoomsController < ApplicationController
...
 
   def index
-    @rooms = Room.page(params[:page]).per Settings.show_rooms
+    @rooms = Room.all
+    if params[:category_id].present?
+      @rooms = @rooms.where(category_id: params[:category_id])
+    end
+    if params[:floor].present?
+      @rooms = @rooms.where(floor: params[:floor])
+    end
```
Comment (mdmahmudurrahman) [inline app/controllers/rooms_controller.rb]: Please write a method in the model for searching rooms and then call it.

File: app/views/layouts/_search_rooms.html.erb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<%= form_tag rooms_path,  remote: true, method: :get, id: "search-rooms" do %>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_search_rooms.html.erb]: Please remove unnecessary white space

File: db/seeds.rb [hunk 5]
```diff
@@ -43,7 +43,7 @@
   user_id = Random.rand(50) + 1
   start_time = Time.zone.now.days_ago(3)
   end_time = Time.zone.now
-  status = Random.rand(5) + 1
+  status = Random.rand(2)
```
Comment (mdmahmudurrahman) [inline db/seeds.rb]: Remove ()

File: db/seeds.rb [hunk 6]
```diff
@@ -26,7 +26,7 @@
 40.times do
   label = Faker::Lorem.characters(5)
   floor = Random.rand(15) + 1
-  status = Random.rand(5) + 1
+  status = Random.rand(3)
```
Comment (mdmahmudurrahman) [inline db/seeds.rb]: Remove ()

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by phucynwa** [comment]
ready

File: app/models/room.rb [hunk 7]
```diff
@@ -8,4 +8,15 @@ class Room < ApplicationRecord
 
   validates :label, :floor, :status, presence: true
   validates :floor, numericality: {greater_than: Settings.zero}
+
+  scope :by_category_id, ->(category_id){where category_id: category_id}
+  scope :by_floor, ->(floor){where floor: floor}
+
+  def self.filter filtering_params
```
Comment (mdmahmudurrahman) [inline app/models/room.rb]: Use `class << self` block to declare class method

File: app/models/room.rb [hunk 8]
```diff
@@ -8,4 +8,15 @@ class Room < ApplicationRecord
...
+    results = Room.all
```
Comment (mdmahmudurrahman) [inline app/models/room.rb]: Do we need this (?)

**Comment by mdmahmudurrahman** [comment]
commented
2 more comment.

**Comment by phucynwa** [comment]
ready

File: app/controllers/rooms_controller.rb [hunk 9]
```diff
@@ -2,7 +2,12 @@ class RoomsController < ApplicationController
   before_action :load_room, only: :show
 
   def index
-    @rooms = Room.page(params[:page]).per Settings.show_rooms
+    @rooms = Room.filter(params.slice(:category_id, :floor))
+      .page(params[:page]).per Settings.show_rooms
```
Comment (mdmahmudurrahman) [inline app/controllers/rooms_controller.rb]: In the model I can see you have defined 2 scope get the room by category and floor. You can use those scopes. 
```ruby
@rooms = Room.by_category_id(params[:category_id]).by_floor(params[:floor])
  .page(params[:page]).per Settings.show_rooms
```

File: app/models/room.rb [hunk 10]
```diff
@@ -8,4 +8,17 @@ class Room < ApplicationRecord
 
   validates :label, :floor, :status, presence: true
   validates :floor, numericality: {greater_than: Settings.zero}
+
+  scope :by_category_id, ->(category_id){where category_id: category_id}
+  scope :by_floor, ->(floor){where floor: floor}
+
+  class << self
+    def filter filtering_params
```
Comment (mdmahmudurrahman) [inline app/models/room.rb]: If used above scopes, you can remove this method

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by phucynwa** [comment]
ready

**Comment by mdmahmudurrahman** [comment]
commented
Please resolve the conflicts

**Comment by phucynwa** [comment]
ready

**Comment by mdmahmudurrahman** [comment]
commented
This pr can be (merged)
CC: [To:2076564] Nguyen Huy Hung C (93)
CC: [To:1818111] Nguyen Thi Huong (94)

**Pull request merged**